### PR TITLE
[bug 1159483] Related documents for KB articles

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: kitsune/sumo/static/js/ui.js [hunk 1]
```diff
@@ -36,23 +36,35 @@
...
+        $target.hide();
       }
     });
 
     $('[data-close-memory="remember"]').each(function() {
-      var id = $(this).data('close-id');
+      var $this = $(this);
+      var id = $this.data('close-id');
       if (id) {
         if (localStorage.getItem(id + '.closed') === 'true') {
```
Comment (mythmon) [inline kitsune/sumo/static/js/ui.js]: I think this needs to be wrapped in a check for localStorage existing.

File: kitsune/sumo/static/js/wiki_search.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      createResultsList();
+    }
+    $resultsList.html(_.render('wiki-search-results.html', data));
+  }
+
+  function handleSearch() {
+    var $this = $(this);
+    if ($this.val().length === 0) {
+      if (searchTimeout) {
+        window.clearTimeout(searchTimeout);
```
Comment (mythmon) [inline kitsune/sumo/static/js/wiki_search.js]: I think it is safe to `clearTimeout(undefined)`, so I don't think you need to have the if statement.

File: kitsune/sumo/static/js/wiki_search.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+(function($, _) {
```
Comment (mythmon) [inline kitsune/sumo/static/js/wiki_search.js]: Should you be calling `k.nunjucksEnv` `_`? That is pretty confusing considering that is also Underscore and usual name.

File: kitsune/sumo/static/less/wiki.less [hunk 4]
```diff
@@ -615,6 +615,48 @@ textarea[name="content"] {
...
+      line-height: 16px;
+      margin-top: -8px;
+      position: absolute;
+      right: 5px;
+      text-align: center;
+      top: 50%;
+      width: 16px;
+
+      &:before {
+        content: '\2715';
```
Comment (mythmon) [inline kitsune/sumo/static/less/wiki.less]: Can you add a comment with what this character is?

File: kitsune/wiki/migrations/0003_auto_20150506_2316.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (mythmon) [inline kitsune/wiki/migrations/0003_auto_20150506_2316.py]: Can you either name the migration to explain what it does, add a module level docstring explaining what it does, or both? I think that is a good habit to  be in for the new migration system.

File: kitsune/sumo/static/js/wiki_search.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    $resultsList.html(_.render('wiki-search-results.html', data));
+  }
+
+  function handleSearch() {
+    var $this = $(this);
+    if ($this.val().length === 0) {
+      if (searchTimeout) {
+        window.clearTimeout(searchTimeout);
+      }
+      $resultsList.html('');
```
Comment (mythmon) [inline kitsune/sumo/static/js/wiki_search.js]: Sometimes I get `$resultsList is undefined` from this line. I think mainly when the window closes right after I make a search.

**Comment by mythmon** [comment]
This works well. Visually it is a little confusing since there isn't a lot of contrast between the results list and the other UI elements on the page. Maybe shifting it left or right a bit will help that? or add a drop shadow? We've done that with other floating parts of the site before.

**Comment by rehandalal** [comment]
updated ^

**Comment by mythmon** [comment]
Looks good!

**Pull request merged**